docs/tutorial/index.md

```diff
@@ -0,0 +1,171 @@
+juju integrate kyuubi-k8s data-integrator
+```
+
+> **Note**: You may use a different object storage, such as `azure-storage-integrator`.
+
+## Access Charmed Apache Kyuubi K8s
+
+Get the JDBC endpoint and its credentials with the following command:
+
+```shell
+juju run data-integrator/0 get-credentials
+```
+
+```yaml
+kyuubi:
+  data: '{"database": "test", "external-node-connectivity": "true", "provided-secrets":
+    "[\"mtls-cert\"]", "requested-secrets": "[\"username\", \"password\", \"tls\",
+    \"tls-ca\", \"uris\", \"read-only-uris\"]"}'
+  database: test
+  endpoints: 10.64.140.43:10009
+  password: 31rwWzk8wpnhoZvU
+  tls: "False"
+  uris: jdbc:hive2://10.64.140.43:10009/
+  username: relation_id_15
+  version: 1.10.2
+ok: "True"
+```
+
+You may use the endpoint with a JDBC-compliant client, such as `beeline`.
+
+```{note}
+We recommend using the [spark-client](https://snapcraft.io/spark-client) snap, which exposes a beeline client under the `spark-client.beeline` command.
+```
+
+```shell
+spark-client.beeline -u "jdbc:hive2://10.64.140.43:10009/" -n relation_id_15 -p 31rwWzk8wpnhoZvU
+```
+
+## Enable encryption with TLS
+
+[Transport Layer Security (TLS)](https://en.wikipedia.org/wiki/Transport_Layer_Security) is a protocol used to encrypt data exchanged between two applications.
+Essentially, it secures data transmitted over a network.
+
+Typically, enabling TLS between a highly available database and client/server applications requires a high level of expertise.
+This has all been encoded into Charmed Apache Kyuubi K8s so that configuring TLS requires minimal effort on your end.
+
+TLS is enabled by integrating Charmed Apache Kyuubi K8s with the [Self-signed certificates charm](https://charmhub.io/self-signed-certificates).
+This charm centralises TLS certificate management consistently and handles operations like providing, requesting, and renewing TLS certificates.
+
+```{caution}
+**[Self-signed certificates](https://en.wikipedia.org/wiki/Self-signed_certificate) are not recommended for a production environment.**
+
+Check [this guide](https://discourse.charmhub.io/t/security-with-x-509-certificates/11664) for an overview of the TLS certificates charms available.
+```
+
+Before enabling TLS on Charmed Apache Kyuubi K8s, we must deploy the `self-signed-certificates` charm:
+
+```shell
+juju deploy self-signed-certificates --config ca-common-name="Tutorial CA"
+```
+
+Wait for the charm settle into an `active/idle` state, as shown by the `juju status`.
+
+To enable TLS on Charmed Apache Kyuubi K8s, relate the `kyuubi-k8s` charm with the `self-signed-certificates` charm:
+
+```shell
+juju relate kyuubi-k8s self-signed-certificates
+```
+
+After the charms settle into `active/idle` states, the Charmed Apache Kyuubi K8s endpoint should now accept encrypted traffic.
+This can be tested by requesting the server certificate using `openssl`:
+
+```shell
+openssl s_client -showcerts -connect 10.64.140.43:10009 < /dev/null
```

Comment:
Question: Can we add a guidance here where to get the address for this command?

Reply:
Switched the commands in bdb99ef2b224ea905b852941fa857653b7fff432 so that the relation between the steps is more apparent + inserted a one liner to get the endpoint